scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -160,8 +160,8 @@ def load(self):
 
             self.load_from_file(last_run, filter_platform=connected_list)
             self.selected_platforms = set(p.platform.name for p in self.instances.values())
-        else:
-            self.apply_filters()
+
+        self.apply_filters()
```

Comment:
Not sure how all of this will work, twister gets a pre-deteremined and set testplan, why does not to apply filters again? I know there might be edge cases of running twister on one host and loading the plan somewhere else which might yield different results, but I think this can be addressed using other means, without changing how this workflow fundemntaly works. Try this:

```
./scripts/twister -e kernel -T tests/kernel/semaphore/
```
Nothing will be built or run, all good. Now put this in a test plan.

```
./scripts/twister -e kernel -T tests/kernel/sempahore --save-tests plan.json
```
then

```
./scripts/twister --load-tests plan.json
```

Tests will run, although the plan says otherwise.

Reply:
> Are you referring to these failures in clang build on non-native_posix platforms, even though they were not in the plan? Which was an issue with my PR, that I asked for help thinking they are CI based? Thanks for pointing me out. I believe it is fixed now. Like I said, now apply_filters() is not adding new instances if there were already existing ones.

no, I am not. 

Just try the above, a testplan which is not supposed to run anything will be completely ignored and tests will run because the original filters will be ignored if you apply the filter again without the actual filter being supplied on the command line.

Reply:
You're right, I missed it... This happened since " it doesn't add new instances during apply_filters() if the instances were already existing. " However, if an instance was existing in the test plan it was evaluated whatever was its status. I agree it was incorrect. 
I fixed it to ignore the existing ones with `status=filtered`. 
I still think the workflow I am proposing is beneficial. As another advantage it allows tuning of the received test plan to limit it furtherly. E.g. `./scripts/twister  -T tests/kernel/semaphore/ --save-tests plan.json` and than `./scripts/twister --load-tests plan.json -e kernel`. Maybe the example is not the most useful one, but shows the idea. 

BTW I wonder if "filtered" tests should be in the test plan? Plan files can get very heavy, and filtered cases can burden it quit a lot. Maybe they should be omitted by default from the plan and stored in a separate file when `--store-filtered <path>` is used?